backend/src/api/services/question/question.service.ts

```diff
@@ -46,14 +46,27 @@ export const getSidebarMetadata = async (studentID: string) => {
 };
 
 export const updateUserBookmark = async (studentID: string, questionID: string) => {
-    const students: [StudentProgress] = await loadData('progress.json');
-    const student = students.find(student => student.studentID === studentID);
-    if (!student) {
-        throw new Error('Student not found');
-    }
-    student.bookMark.push(questionID);
+    try {
+        // Check if the question exists
+        const question = await QuestionBaseModel.findById(questionID);
+        if (!question) {
+            throw new Error('Question not found');
+        }
 
-    return student;
+        // Find the student
+        const student = await StudentProgressModel.findOne({ studentID });
+        if (!student) {
+            throw new Error('Student not found');
+        }
+
+        // Update user bookmark
+        student.bookMark.push(questionID);
+        await student.save();
+        
+        return student;
+    } catch (error: any) {
+        throw new Error('Error updating user bookmark: ' + error.message);
+    }
```

Comment:
this try + catch is not needed since the the error will automatically be caught in controller if you look at the controller code there is a try + catch there to catch any errors.